raft-tech/TANF-app
# Issue 691: As a developer, I would like the app to hot reload when running locally

### Summary of Changes
This PR updates the `Dockerfile.local` and `docker-compose.yml`, and creates new `docker-compose.local.yml` and `docker-compose.dev.yml` for the frontend, enabling support for hot reloading by using the built in `react-scripts start` command.

_Addresses issue #691
_Acceptance criteria as stated in the issue_

### How to Test
_List the steps to test the PR_

- Checkout this branch `ops/691-docker-hot-reload` and [run the backend](https://github.com/raft-tech/TANF-app#running-the-application-locally).
- `cd` into the `tdrs-frontend` directory, and rebuild its containers:
	- Remove any currently running containers: `docker-compose down --remove-orphans`
- Run the frontend containers: ` docker-compose -f docker-compose.yml -f docker-compose.local.yml up --build`
- Make changes to any frontend code:
	- I suggest updating the header in `SplashPage.jsx`
- Ensure a reload was triggered in the window running the app.
- Ensure your changes are visible in the app.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

  - [x] `docker-compose up` should serve the app at `http://localhost:3000` as before (this should not have changed)
  - [x] Local frontend code changes should trigger a hot reload
  - [x] Changes should be visible in UI after reload

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*



### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*



### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*



### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*



### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

	- No security issues detected
